### ISSUE TITLE
As developer I want to have a complete build during installation

### ISSUE THREAD

**Issue by franz-josef-kaiser** [wecodemore/wcm-avatar#27]
Currently we have a Git hook to run the _NPM_ build process, which then uses bash scripts to call the proper commands. We also have a `.po`-file generator that sits in the root. All this should better be summoned and run during a Composer [install script](https://getcomposer.org/doc/articles/scripts.md) as this is when the package gets pulled and set up. In addition, we should continue to ship the assets in minified versions to have a fallback.

**Comment by kraftner** [comment]
I think the whole build process is broken:

[package.json#L17](https://github.com/wecodemore/wcm-avatar/blob/58caa432ab5637aabbcc8bd7dfd9a90e5337b0bc/package.json#L17) and [scripts/scripts](https://github.com/wecodemore/wcm-avatar/blob/58caa432ab5637aabbcc8bd7dfd9a90e5337b0bc/scripts/scripts) don't match.
After that it still doesn't work because something is wrong with [scripts/scripts#L2](https://github.com/wecodemore/wcm-avatar/blob/58caa432ab5637aabbcc8bd7dfd9a90e5337b0bc/scripts/scripts#L2)

Also the versions used look old and I'm note sure of the packages themselves.

So yeah, this all needs a major overhaul. :smile:

**Comment by franz-josef-kaiser** [comment]
@kraftner should be working again. Use `npm run build` to trigger a build. I am not really sure if the `watch` process works. The last version used to just call `npm run build`, which triggered endlessly. The current version just includes the commands from the `js` and the `css` files. While this might work better, the file location is relative to the location from where you call the script – it just assumes you are in the project root when doing so. This might fail, please give this a test run. The branch to test is `issue-27-assets`.

**Comment by kraftner** [comment]
Yeah, works again. I also noticed that as it seems up until now there was a mismatch between src and concat/minified files like e.g. [models.js](https://github.com/wecodemore/wcm-avatar/blob/58caa432ab5637aabbcc8bd7dfd9a90e5337b0bc/assets/src/logo/models.js#L40-L51) and [all.js](https://github.com/wecodemore/wcm-avatar/blob/58caa432ab5637aabbcc8bd7dfd9a90e5337b0bc/assets/src/logo/all.js#L40-L49)

I'll add a rebuild of the assets to the issue branch and merge this tomorrow.

**Comment by franz-josef-kaiser** [comment]
Thanks @kraftner – Please do not forget about #48

**Comment by kraftner** [comment]
@franz-josef-kaiser Is this good enough in your opinion or do you want to keep it open to further improve it?
